File: lab1.cpp


#include <iostream>
#include <cmath>
using namespace std;

int main() {

   
    // TASK 1 
   

    cout << "----- TASK 1 -----" << endl;

    string name;
    string studentID;

    cout << "What is your name?" << endl;
    cin >> name;

    cout << "Hello " << name << "." << endl;

    cout << "What is your Student ID?" << endl;
    cin >> studentID;

    cout << "Your ID is " << studentID << "." << endl;


    
    // TASK 2
    

    cout << "\n----- TASK 2 -----" << endl;

    int totalSeconds;

    cout << "Enter total seconds: ";
    cin >> totalSeconds;

    int hours = totalSeconds / 3600;
    int remainingSeconds = totalSeconds % 3600;
    int minutes = remainingSeconds / 60;
    int seconds = remainingSeconds % 60;

    cout << totalSeconds << " seconds is "
         << hours << " hours, "
         << minutes << " minutes, and "
         << seconds << " seconds." << endl;


    
    // TASK 3 
   

    cout << "\n----- TASK 3 -----" << endl;

    double x1, y1, x2, y2;

    cout << "Enter x1: ";
    cin >> x1;

    cout << "Enter y1: ";
    cin >> y1;

    cout << "Enter x2: ";
    cin >> x2;

    cout << "Enter y2: ";
    cin >> y2;

    double distance = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));

    cout << "The distance between the points is " << distance << endl;


    
    // TASK 4 
    

    cout << "\n----- TASK 4 -----" << endl;

    cout << "*******\n *****\n  ***\n   *" << endl;


    return 0;
}

File: lab1.py
import math



# TASK 1


print("----- TASK 1 -----")

name = input("What is your name?\n")
print(f"Hello {name}.")

student_id = input("What is your Student ID?\n")
print(f"Your ID is {student_id}.")



# TASK 2 - Time Logic


print("\n----- TASK 2 -----")

total_seconds = int(input("Enter total seconds:\n"))

hours = total_seconds // 3600
remaining_seconds = total_seconds % 3600
minutes = remaining_seconds // 60
seconds = remaining_seconds % 60

print(f"{total_seconds} seconds is {hours} hours, {minutes} minutes, and {seconds} seconds.")



# TASK 3 


print("\n----- TASK 3 -----")

x1 = float(input("Enter x1: "))
y1 = float(input("Enter y1: "))
x2 = float(input("Enter x2: "))
y2 = float(input("Enter y2: "))

distance = math.sqrt((x2 - x1)**2 + (y2 - y1)**2)

print(f"The distance between the points is {distance}")



# TASK 4 


print("\n----- TASK 4 -----")

print("*******\n *****\n  ***\n   *")
